packages/eas-cli/src/commands/channel/rollout.ts

```diff
@@ -0,0 +1,295 @@
+      throw new Error('This CLI cannot handle branch mappings with more than 2 branches.');
+    }
+
+    if (isRollout && branchName && !endFlag) {
+      throw new Error(
+        `There is a rollout in progress. You can only either edit the rollout percent or 'end' it.`
+      );
+    }
+
+    let newChannelInfo, logMessage;
+    if (!isRollout) {
+      // start a rollout
+      // this will likely be removed as more commands fill out
+      if (!branchName) {
+        const validationMessage = 'A branch must be specified.';
+        if (jsonFlag) {
+          throw new Error(validationMessage);
+        }
+        ({ name: branchName } = await promptAsync({
+          type: 'text',
+          name: 'name',
+          message: `Select a branch to rollout onto ${channelName}`,
+          validate: value => (value ? true : validationMessage),
+        }));
+      }
+      const branch = await getBranchByNameAsync({ appId: projectId, name: branchName! });
+
+      if (percent === undefined) {
+        const promptMessage = `What percent of users should be directed to the branch ${branchName}?`;
+        const validationMessage =
+          'The rollout percentage must be an integer between 0 and 99 inclusive.';
+        if (jsonFlag) {
+          throw new Error(validationMessage);
+        }
+        percent = await getPercentAsync({ promptMessage, validationMessage });
+      }
+
+      const newBranchMapping = {
+        version: 0,
+        data: [
+          {
+            branchId: branch.id,
+            branchMappingLogic: {
+              operand: percent! / 100,
+              clientKey: 'rolloutToken',
+              branchMappingOperator: 'hash_lt',
+            },
+          },
+          currentBranchMapping.data[0],
+        ],
+      };
+      newChannelInfo = await updateChannelBranchMappingAsync({
+        channelId: getUpdateChannelByNameForAppResult.app?.byId.updateChannelByName.id!,
+        branchMapping: JSON.stringify(newBranchMapping),
+      });
+
+      const oldBranchId = currentBranchMapping.data[0].branchId;
+      const oldBranch = getUpdateChannelByNameForAppResult.app?.byId.updateChannelByName.updateBranches.filter(
+        branch => branch.id === oldBranchId
+      )[0];
+      if (!oldBranch) {
+        throw new Error(
+          `Branch mapping is missing its only branch for channel "${channelName}" on app "@${accountName}/${slug}"`
+        );
+      }
+
+      logMessage = `️Started a rollout of branch ${chalk.bold(
+        branchName
+      )} onto channel ${chalk.bold(channelName!)}! ${chalk.bold(
+        percent
+      )}% of users will be directed to ${chalk.bold(branchName)}, ${chalk.bold(
+        100 - percent!
+      )}% to ${oldBranch.name}.`;
+    } else if (!endFlag) {
+      if (jsonFlag && !percent) {
+        throw new Error(
+          'A rollout is already in progress. You can either end it with the --end flag or modify it with the --percent flag.'
+        );
+      }
```

Comment:
delet in favor of: https://github.com/expo/eas-cli/pull/290/files#diff-5e715a56413f58e4e51aad67353e66469536e961c39884dd972b572a3d847f2dR206